### PR TITLE
Merge changes from upstream.

### PR DESCRIPTION
**Pull request by knz**

### CODE REVIEW

File: cockroachdb/README.rst [hunk 1]
```diff
@@ -77,6 +84,28 @@ added two or more times; that all known-ok additions are indeed
...
+
+Each node may report ok, the operation is known to have succeeded;
+fail, the operation is known to have failed; and unknown otherwise
+(e.g. the connection dropped before the answer could be read).
+
+At the end, a monotonic checker validates that no value was
+added two or more times; that all known-ok additions are indeed
+present in the table; that all values are in order.
+
+Test detais: monotonic with clock skews
```
Comment (tbg) [inline cockroachdb/README.rst]: s/detais/details/

File: cockroachdb/README.rst [hunk 2]
```diff
@@ -77,6 +84,28 @@ added two or more times; that all known-ok additions are indeed
...
+---------------------------------------
+
+Same test setup as above; however here the nemesis insert random clock
+skews of +/- 100ms on each node intead of partitions.
```
Comment (tbg) [inline cockroachdb/README.rst]: s/intead/instead/

File: cockroachdb/README.rst [hunk 3]
```diff
@@ -77,6 +84,28 @@ added two or more times; that all known-ok additions are indeed
...
+
+Concurrently, a nemesis partitions the network between the nodes randomly.
+
+Each node may report ok, the operation is known to have succeeded;
+fail, the operation is known to have failed; and unknown otherwise
+(e.g. the connection dropped before the answer could be read).
+
+At the end, a monotonic checker validates that no value was
+added two or more times; that all known-ok additions are indeed
+present in the table; that all values are in order.
```
Comment (tbg) [inline cockroachdb/README.rst]: it's not clear what "in order" means.

File: jepsen/src/jepsen/util.clj [hunk 4]
```diff
@@ -188,6 +189,34 @@
              (recur))
          res#))))
 
+(defrecord Retry [bindings])
+
+(defmacro with-retry
+  "It's really fucking inconvenient not being able to recur from within (catch)
```
Comment (tbg) [inline jepsen/src/jepsen/util.clj:195]: s/fucking//

Comment (knz) [inline jepsen/src/jepsen/util.clj:195]: On 02/15/2016 21:50, Tobias Schottdorf wrote:

> In jepsen/src/jepsen/util.clj
> https://github.com/cockroachdb/jepsen/pull/2#discussion_r52942165:

That's a change from upstream. :)

**Comment by tbg** [comment]
I think getting master rebased first is better for the review, since now there are a bunch of unrelated upstream changes in the diff. I think it's fine if you merge your work in progress; reviewing this in detail is difficult without testing it. I'm planning to, but that's best done in the context of an anomaly.
LGTM

**Pull request merged**